src/reducers/versions.spec.tsx

```diff
@@ -175,23 +176,68 @@ describe(__filename, () => {
   });
 
   describe('getVersionFile', () => {
+    /* eslint-disable @typescript-eslint/camelcase */
     it('returns a version file', () => {
+      const mimeType = 'mime/type';
       const path = 'test.js';
+      const type = 'text';
+      const version = {
+        ...fakeVersion,
+        file: {
+          ...fakeVersionFile,
+          entries: {
+            [path]: {
+              ...fakeVersionEntry,
+              mime_category: type as VersionEntryType,
+              mimetype: mimeType,
+              path,
+            },
+          },
+          selected_file: path,
+        },
+      };
+      const state = reducer(undefined, actions.loadVersionInfo({ version }));
+
+      expect(getVersionFile(state, version.id, path)).toEqual({
+        ...createInternalVersionFile(version.file),
+        mimeType,
+        type,
+      });
+    });
+    /* eslint-enable @typescript-eslint/camelcase */
+
+    it('returns undefined if there is no version found', () => {
+      const state = initialState;
+
+      expect(getVersionFile(state, 1, 'some-file-name.js')).toEqual(undefined);
+    });
+
+    it('returns undefined if there is no file for the path', () => {
       const version = fakeVersion;
       const state = reducer(
         undefined,
-        actions.loadVersionFile({ path, version }),
+        actions.loadVersionInfo({ version: fakeVersion }),
       );
 
-      expect(getVersionFile(state, version.id, path)).toEqual(
-        createInternalVersionFile(version.file),
-      );
+      expect(
+        getVersionFile(state, version.id, `${version.file.selected_file}-1`),
```

Comment:
is `-1` used to pass an unknown file? I think passing `unknown-file` would be enough.

Reply:
Yes, the intention is to pass a path for a file that hasn't been loaded.

I was trying to be explicit in the test that the path we're requesting is definitely not the one that's already loaded. The test loads a version with default values, rather than with an explicit path, because the latter requires more set up. So while *we* know that `unknown-file` is a path that doesn't match what's loaded, that's not explicit in the test.

Does that all make sense? I just don't like to rely on a knowledge of what the default values are in a test. Using `${version.file.selected_file}-1` is explicitly using a value that does not match what has been loaded.

Reply:
I find it more difficult to understand than hard-coding a value that is self-explanatory but 🤷‍♂️ . Maybe add another suffix than `-1` then.

Reply:
Fair enough. I'll just use a self-explanatory string, as you suggest.